File: apps/android-demo/app/src/main/java/com/gsplat/demo/NativeBridge.java
package com.gsplat.demo;

public final class NativeBridge {
  static {
    System.loadLibrary("gsplat_jni");
  }

  private NativeBridge() {}

  public static native int versionMajor();

  public static native int versionMinor();

  public static native int runFfiSmoke(String datasetPath);
}

File: apps/android-demo/app/src/main/kotlin/com/gsplat/demo/NativeBridge.kt
package com.gsplat.demo

import android.view.Surface

object NativeBridge {
    init {
        System.loadLibrary("gsplat_jni")
    }

    @JvmStatic
    external fun versionMajor(): Int

    @JvmStatic
    external fun versionMinor(): Int

    @JvmStatic
    external fun runFfiSmoke(datasetPath: String): Int

    @JvmStatic
    external fun createSurfaceRenderer(
        surface: Surface,
        datasetPath: String,
        width: Int,
        height: Int
    ): Long

    @JvmStatic
    external fun resizeSurfaceRenderer(nativeHandle: Long, width: Int, height: Int): Int

    @JvmStatic
    external fun renderSurfaceFrame(nativeHandle: Long): Int

    @JvmStatic
    external fun getSurfaceStats(nativeHandle: Long, outStats: LongArray): Int

    @JvmStatic
    external fun destroySurfaceRenderer(nativeHandle: Long)
}
